app/Http/Controllers/Admin/QrcodeController.php

```diff
@@ -28,4 +30,43 @@ public function index()
             ->paginate(config('define.page_length'));
         return view('backend.qrcodes.index', compact('qrcodes'));
     }
+
+    /**
+     * Export QR Codes Not Printed.
+     *
+     * @return mixed
+     */
+    public function exportCSV()
+    {
+        DB::beginTransaction();
+        try {
+            $fields = [
+                'qrcodes.id',
+                'books.name',
+                'books.author',
+                DB::raw("CONCAT(qrcodes.prefix, qrcodes.code_id) AS QR_Codes")
+            ];
+            $datas = QrCode::select($fields)->QRCodesNotPrinted()->join('books', 'qrcodes.book_id', 'books.id')->get();
+            
+            if ($datas->isNotEmpty()) {
```

Comment:
xem lại có cần thiết dùng `if` ở đây ko em?
- trong vòng lặp for, nếu $datas empty thì nó cũng ko vào đó.
- lúc export csv, thì có lẽ nó cũng ko xuất file, hoặc có thể xuất file rỗng rồi

Reply:
dạ, em dùng if để kiểm tra nếu danh sách QR Codes mà không empty thì mới cho phép in, còn nếu  danh sách empty thì sẽ không cho phép export và sẽ hiển thị thông báo lỗi.

Reply:
hiểu lun, nhưng có cần ko, em có thể thử ko if xem nó có đúng ko

Reply:
dạ, em dùng `if` để check nếu dữ liệu `not empty` thì sẽ không cho export nữa., nếu không có `if` thì lần export sau nó sẽ vẫn export ra 1 file dữ liệu trống